core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -84,30 +143,157 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
-          this.TAB_NAME_QUESTION_REVIEWER];
-      }
-      if (this.isQuestionCoordinator && this.isTranslationCoordinator) {
-        this.CONTRIBUTION_TYPES = [this.TAB_NAME_TRANSLATION_SUBMITTER,
-          this.TAB_NAME_TRANSLATION_REVIEWER,
+        this.CONTRIBUTION_TYPES.push(
           this.TAB_NAME_QUESTION_SUBMITTER,
-          this.TAB_NAME_QUESTION_REVIEWER];
+          this.TAB_NAME_QUESTION_REVIEWER);
       }
 
+      this.contributorDashboardAdminStatsBackendApiService
+        .fetchAssignedLanguageIds(username).then(
+          response => {
+            this.languageChoices = this.languageChoices.filter((
+                languageItem) =>
+              response.includes(languageItem.id));
+            this.selectedLanguage.language = (
+              this.languageChoices[0].language);
+            this.selectedLanguage.id = this.languageChoices[0].id;
+            this.translationReviewersCount = (
+              this.translationReviewersCountByLanguage[
+                this.selectedLanguage.id]);
+          });
+
+      this.filter = new ContributorAdminDashboardFilter(
+        [],
+        this.selectedLanguage.id,
+        null,
+        this.selectedLastActivity);
+
       this.updateSelectedContributionType(this.CONTRIBUTION_TYPES[0]);
+
+      this.loadingMessage = '';
       this.changeDetectorRef.detectChanges();
     });
 
+    this.lastActivity = [0, 7, 30, 90];
+
     this.contributorDashboardAdminStatsBackendApiService
-      .fetchCommunityStats().then(
-        (response) => {
-          this.translationReviewersCount = response.translation_reviewers_count;
-          this.questionReviewersCount = response.question_reviewers_count;
-        });
+      .fetchTopicChoices().then(
+        response => {
+          this.topics = response;
+          response.map((
+              topic) =>
+            this.allTopicNames.push(topic.topic));
+          this.applyTopicFilter();
+        }
+      );
+  }
+
+  toggleLanguageDropdown(): void {
+    this.languageDropdownShown = !this.languageDropdownShown;
+  }
+
+  toggleActivityDropdown(): void {
+    this.activityDropdownShown = !this.activityDropdownShown;
+  }
+
+  applyTopicFilter(): void {
+    this.selectedTopics.ids = this.selectedTopics.topics.map(
+      selectedTopic => {
+        const matchingTopic = this.topics.find(
+          topicChoice => topicChoice.topic === selectedTopic);
+        return matchingTopic ? matchingTopic.id : '';
+      });
+    this.filter = new ContributorAdminDashboardFilter(
+      this.selectedTopics.ids,
+      this.selectedLanguage.id,
+      null,
+      this.selectedLastActivity);
+  }
+
+  selectLanguage(language: string): void {
+    const currentOption: LanguageChoice = this.languageChoices.find(
+      option => option.language === language) || {
+      language: 'English',
+      id: 'en'
+    };
+    this.selectedLanguage.language = currentOption.language;
+    this.selectedLanguage.id = currentOption.id;
```

Comment:
Same here just assign `this.selectedLanguage` in one line.

Reply:
Sg. Fixed.